middleware/nethttp/abi.go

```diff
@@ -0,0 +1,91 @@
+package wasm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+
+	httpwasm "github.com/http-wasm/http-wasm-host-go"
+)
+
+func compileHost(ctx context.Context, r wazero.Runtime, logger httpwasm.Logger) (wazero.CompiledModule, error) {
+	h := &host{logger: logger}
+	if compiled, err := r.NewHostModuleBuilder("http").
+		ExportFunction("log", h.log, "log", "ptr", "size").
+		ExportFunction("next", h.next, "next").
+		Compile(ctx); err != nil {
+		return nil, fmt.Errorf("wasm: error compiling host: %w", err)
+	} else {
+		return compiled, nil
+	}
+}
+
+func compileGuest(ctx context.Context, r wazero.Runtime, wasm []byte) (wazero.CompiledModule, error) {
```

Comment:
when we add fasthttp, a lot of this stuff would be put into an internal package and re-used. just not doing that now because it would be more distracting.